lib/spandx/core/report.rb

```diff
@@ -13,11 +13,11 @@ class Report
       }.freeze
 
       def initialize
-        @dependencies = []
+        @dependencies = Set.new
```

Comment:
@mokhan 

If I understand this correctly we are putting everything to set after all lookup is done. I would prefer using set when we extracting dependencies because we could eliminate duplicates which means we could eliminate unnecessary lookups . Let me know if I am overlooking something

Reply:
Oh yeah, good catch. I moved it up to the report because I found that a duplicate dependency was detected in the bundler lockfile parser as well. By moving it to the report we prevent duplicates from appear in report output. I overlooked that fact that this could cause an additional outbound network request. I will restore this in the yarn lockfile parser.